main.js

```diff
@@ -21,15 +21,62 @@ for (let i = 0; i < arr.length; i++) {
 console.log(result);
 //part 4
 let app = [2, 5, 8, 15, 0, 6, 20, 3];
-for (let i = 0; i < app.length; i++){
-    if (i > 5 && i < 10){
-        console.log(app);
-    }
+for (let i = 0; i < app.length; i++) {
+  if (i > 5 && i < 10) {
+    console.log(app);
+  }
 }
 //part 5
 let num = [2, 5, 8, 15, 0, 6, 20, 3];
 for (let i = 0; i < num.length; i++) {
-    if (num[i] % 2 === 0) {
-        console.log(num[i]);
-    }
-}
\ No newline at end of file
+  if (num[i] % 2 === 0) {
+    console.log(num[i]);
+  }
+}
+//lesson-3
+//part 1
+function palindrome(str) {
+  str = str.toLowerCase().replace(/\s/g, '');
+  return str === str.split('').reverse().join('');
+}
+console.log(palindrome('racecar')); //true
+console.log(palindrome('magazine')); //false
+console.log(palindrome('Bob')); //true
+console.log(palindrome('Madam Im Adam')); //true
+//part 2
+//Возвращение минимального значения
+function min(a, b) {
+  if (a < b) {
+    return a;
+  } else {
+    return b;
+  }
+}
+/* function min(a, b) {
+    return a < b ? a : b; --тернарный вариант
+}*/
+console.log(min(2, 5)); // вернёт 2
+//Возвращение максимального значения
+function max(a, y) {
+  if (a > y) {
+    return a;
+  } else {
+    return y;
+  }
+}
+/* function max(a, b) {
+    return a > b ? a : b; --тернарный вариант
+}*/
+console.log(max(12, 100)); //вернёт 100
+//part 3
+const generateArray = (length, max) =>
+  [...new Array(length)].map(() => Math.round(Math.random() * max));
+const numbers = generateArray(10, 100);
+function replaceZero(number) {
+  const numStr = `${number}`;
```

Comment:
This is a very interessting way to change a number on a string. It will be work absolutely correct.
However using String() or .toString() are more **predictable** ways of convertation numbers to strings.
I recommend just to think why String() and .toString()  are more predictable and why their predictable behavior is better